app/controllers/user_posts_controller.rb

```diff
@@ -5,7 +5,66 @@ def index
   end
 
   def show
+    @user, @post = user_post_helper
+  end
+
+  def new
     @user = User.find(params[:user_id])
-    @post = Post.find(params[:post_id])
+    @post = Post.new
+  end
+
+  def create
+    @current_user = User.find(params[:user_id])
+    @post = Post.new(user: @current_user, title: params[:post][:title], text: params[:post][:text])
+    if @post.save
+      flash[:success] = 'Post successfully created'
+      redirect_to "/users/#{@current_user.id}/posts/#{@post.id}"
+    else
+      flash.now[:error] = 'Error: Post could not be created'
+      render :new
+    end
+  end
+
+  def createcomment
+    @user, @post = user_post_helper
+  end
+
+  def submitcomment
+    @user, @post = user_post_helper
+
+    @comment = Comment.create(post_id: @post.id, user_id: @user.id, text: params[:post][:text])
+
+    if @comment.save
+
+      flash[:success] = 'Comment successfully created'
+      redirect_to "/users/#{@user.id}/posts/#{@post.id}"
+    else
+      flash.now[:error] = 'Error: Comment could not be created'
+      render :createcomment
+    end
+  end
```

Comment:
- [ ] Hello Dear, I really like your project. Your project looks great 💯. Thank you for putting in so much work on this project 🥇. However, according to the project requirement, your project is expected to have a `Comments` controller where you have all the actions for creating a comment for a post. Kindly create a `Comments` controller and then move all the actions that are related to creating a comment to the `Comments` controller okay. This is to have a good separation of concern and to fulfill the project requirement. I know that this will not be hard for you to do after seeing all the effort you've put into this project. 👍 Please remember to create the routes and views accordingly.

Reply:
@julie-ify Thanks for this, now my code makes even more sense!